artifacts/definitions/Windows/EventLogs/PowershellScriptblock.yaml

```diff
@@ -0,0 +1,83 @@
+name: Windows.EventLogs.PowershellScriptblock
+description: |
+  This Artifact will search and extract ScriptBlock events (Event ID 4104) from 
+  Powershell-Operational Event Logs.
+  
+  Powershell is commonly used by attackers accross all stages of the attack 
+  lifecycle. A valuable hunt is to search Scriptblock logs for signs of 
+  malicious content.
+  
+  There are several parameter's availible for search leveraging regex.    
+    - SearchStrings enables regex search over scriptblock text field.  
+    - stringWhiteList enables a regex whitelist for scriptblock text field.  
+    - pathWhitelist enables a regex whitelist for path of scriptblock.  
+    - dateAfter enables search for events after this date.  
+    - dateBefore enables search for events before this date.  
+  
+
+author: Matt Green - @mgreen27
+
+parameters:
+  - name: eventLog
+    default: C:\Windows\system32\winevt\logs\Microsoft-Windows-PowerShell%4Operational.evtx
+  - name: searchStrings
+    description: "regex search over scriptblock text field."
+    default: (?i).
+  - name: dateAfter
+    description: "search for events after this date."
+    default: "2000-01-01"
```

Comment:
Have you tried to set the type as type: timestamp 

The GUI will render that as a time selector box and will write the epoch time in there - does it look better?

I think the time guessing code will handle either (epoch or timestamp) so it should not matter to the VQL.

Reply:
added